src/object/clone.ts

```diff
@@ -26,48 +26,103 @@
  * console.log(clonedObj); // { a: 1, b: 'es-toolkit', c: [1, 2, 3] }
  * console.log(clonedObj === obj); // false
  */
-export function clone<T>(obj: T): T {
+export function clone<T>(obj: T): Shallowed<T> {
   if (isPrimitive(obj)) {
-    return obj;
+    return obj as Shallowed<T>;
   }
 
   if (Array.isArray(obj)) {
-    return obj.slice() as T;
+    return obj.slice() as Shallowed<T>;
   }
 
   if (obj instanceof Date) {
-    return new Date(obj.getTime()) as T;
+    return new Date(obj.getTime()) as Shallowed<T>;
   }
 
   if (obj instanceof RegExp) {
-    return new RegExp(obj.source, obj.flags) as T;
+    return new RegExp(obj.source, obj.flags) as Shallowed<T>;
   }
 
   if (obj instanceof Map) {
     const result = new Map();
     for (const [key, value] of obj) {
       result.set(key, value);
     }
-    return result as T;
+    return result as Shallowed<T>;
   }
 
   if (obj instanceof Set) {
     const result = new Set();
     for (const value of obj) {
       result.add(value);
     }
-    return result as T;
+    return result as Shallowed<T>;
   }
 
-  if (typeof obj === "object") {
-    return Object.assign({}, obj) as T;
+  // BINARY DATA
+  if (obj instanceof Uint8Array) return new Uint8Array(obj) as Shallowed<T>;
+  if (obj instanceof Uint8ClampedArray) return new Uint8ClampedArray(obj) as Shallowed<T>;
+  if (obj instanceof Uint16Array) return new Uint16Array(obj) as Shallowed<T>;
+  if (obj instanceof Uint32Array) return new Uint32Array(obj) as Shallowed<T>;
+  if (obj instanceof BigUint64Array) return new BigUint64Array(obj) as Shallowed<T>;
+  if (obj instanceof Int8Array) return new Int8Array(obj) as Shallowed<T>;
+  if (obj instanceof Int16Array) return new Int16Array(obj) as Shallowed<T>;
+  if (obj instanceof Int32Array) return new Int32Array(obj) as Shallowed<T>;
+  if (obj instanceof BigInt64Array) return new BigInt64Array(obj) as Shallowed<T>;
+  if (obj instanceof Float32Array) return new Float32Array(obj) as Shallowed<T>;
+  if (obj instanceof Float64Array) return new Float64Array(obj) as Shallowed<T>;
+  if (obj instanceof ArrayBuffer) return obj.slice(0) as Shallowed<T>;
+  if (obj instanceof SharedArrayBuffer) return obj.slice(0) as Shallowed<T>;
+  if (obj instanceof DataView) return new DataView(obj.buffer.slice(0)) as Shallowed<T>;
+  if (obj instanceof Blob) return new Blob([obj], { type: obj.type }) as Shallowed<T>;
+  if (obj instanceof File) return new File([obj], obj.name, { type: obj.type }) as Shallowed<T>;
```

Comment:
The following line may not work as intended because File extends Blob: 
```suggestion
  if (obj instanceof File) return new File([obj], obj.name, { type: obj.type }) as Shallowed<T>;
  if (obj instanceof Blob) return new Blob([obj], { type: obj.type }) as Shallowed<T>;
```

Reply:
<img width="466" alt="스크린샷 2024-07-17 오전 2 02 35" src="https://github.com/user-attachments/assets/8f554094-9b73-4419-8e1c-c52f28e8dec3">